src/components/UASTViewer.js

```diff
@@ -13,7 +13,13 @@ export const getSearchResults = uast => {
     return null;
   }
 
-  if (Array.isArray(rootNode.n)) {
+  // for search server returns [result, result, result]
+  // in case when all the results are nodes we can ignore wrapper node
+  // and display only results for better UI/UX.
+  //
+  // for values (strings, ints, bool) we have to keep the wrapping node
+  // otherwise it's not a tree and uast-viewer wouldn't be able to display it
+  if (Array.isArray(rootNode.n) && rootNode.n.every(c => c.id)) {
```

Comment:
Probably I'm missing something as I don't know much this part, but I don't understand this. In this condition, you're checking that `rootNode.n` is an array and that each is element has an `id` property that I guess is a proxy for checking that each element is a node. So this part:

>   // in case when all the results are nodes we can ignore wrapper node
  // and display only results for better UI/UX.

is clear. The part that I don't understand is:

>   // for values (strings, ints, bool) we have to keep the wrapping node
  // otherwise it's not a tree and uast-viewer wouldn't be able to display it

Shouldn't you return `rootNode.n` if this `rootNode.n.every(c => c.id)` it's not true?

```
if (Array.isArray(rootNode.n)) {
    if (rootNode.n.every(c => c.id)) {
        return rootNode.n.map(c => c.id);
    } else {
        return rootNode.n;
    }
}
```

Reply:
this function returns root ids, not nodes.
If all the results are nodes it would return ids of them.
Otherwise, it would return `null` and default root id (of the wrapper) would be used.

Reply:
oh ok! Thanks for the explanation!